deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -0,0 +1,317 @@
+        train_dataset.validate_label()
+        test_dataset.validate_label()
+        train_dataset.validate_shared_label(test_dataset)
+        train_dataset.validate_shared_features(test_dataset)
+        train_dataset.validate_shared_categorical_features(test_dataset)
+        validate_model(train_dataset, model)
+
+        task_type = task_type_check(model, train_dataset)
+
+        # If user defined scorers used them, else use a single scorer
+        if self.alternative_scorer:
+            scorer = get_scorers_list(model, train_dataset, self.alternative_scorer, multiclass_avg=True)[0]
+        else:
+            scorer = get_scorer_single(model, train_dataset, multiclass_avg=True)
+
+        cat_features = train_dataset.cat_features
+
+        train_dataset = train_dataset.sample(self.n_samples, random_state=self.random_state)
+        test_dataset = test_dataset.sample(self.n_samples, random_state=self.random_state)
+
+        # Create scoring function, used to calculate the per sample model error
+        if task_type == ModelType.REGRESSION:
+            def scoring_func(dataset):
+                return per_sample_mse(dataset.label_col, model.predict(dataset.features_columns))
+        else:
+            def scoring_func(dataset):
+                return per_sample_binary_cross_entropy(dataset.label_col,
+                                                       model.predict_proba(dataset.features_columns))
+
+        train_scores = scoring_func(train_dataset)
+        test_scores = scoring_func(test_dataset)
+
+        # Create and fit model to predict the per sample error
+        error_model = create_error_model(train_dataset, random_state=self.random_state)
+        error_model.fit(train_dataset.features_columns, y=train_scores)
+
+        # Check if fitted model is good enough
+        error_model_predicted = error_model.predict(test_dataset.features_columns)
+        error_model_score = r2_score(test_scores, error_model_predicted)
+
+        # This check should be ignored if no information gained from the error model (low r2_score)
+        if error_model_score < self.min_error_model_score:
+            raise DeepchecksProcessError(f'Unable to train meaningful error model '
+                                         f'(r^2 score: {format_number(error_model_score)})')
+
+        error_fi = calculate_feature_importance(error_model, test_dataset,
+                                                permutation_kwargs={'random_state': self.random_state})
+        error_fi.sort_values(ascending=False, inplace=True)
+
+        n_samples_display = min(self.n_display_samples, len(test_dataset))
+        display_data = test_dataset.data
+        error_col_name = 'Deepchecks model error'
+        display_error = pd.Series(error_model_predicted, name=error_col_name, index=test_dataset.data.index)
+
+        display = []
+        value = {}
+        self._scorer_name = scorer.name
+        weak_color = '#d74949'
+        ok_color = colors['Test']
```

Comment:
Why not add the weak color to the colors dict? or anyway, put this constant outside?